examples/experimental/dagster-airlift/dagster_airlift/core/sensor/sensor_builder.py

```diff
@@ -307,37 +304,59 @@ def build_synthetic_asset_materializations(
     This also currently does not support dynamic tasks in Airflow, in which case
     the use should instead map at the dag-level granularity.
     """
-    task_instances = airflow_instance.get_task_instance_batch(
-        run_id=dag_run.run_id,
-        dag_id=dag_run.dag_id,
-        task_ids=[task_id for task_id in airflow_data.task_ids_in_dag(dag_run.dag_id)],
-        states=["success"],
-    )
-
-    context.log.info(f"Found {len(task_instances)} task instances for {dag_run.run_id}")
-    context.log.info(f"All task instances {task_instances}")
-
-    check.invariant(
-        len({ti.task_id for ti in task_instances}) == len(task_instances),
-        "Assuming one task instance per task_id for now. Dynamic Airflow tasks not supported.",
-    )
-
     # https://linear.app/dagster-labs/issue/FOU-444/make-sensor-work-with-an-airflow-dag-run-that-has-more-than-1000
     dagster_runs = context.instance.get_runs(
         filters=RunsFilter(tags={DAG_RUN_ID_TAG_KEY: dag_run.run_id}),
         limit=1000,
     )
+    context.log.info(f"Found {len(dagster_runs)} dagster runs for {dag_run.run_id}")
 
     context.log.info(
         f"Airlift Sensor: Found dagster run ids: {[run.run_id for run in dagster_runs]}"
         f" for airflow run id {dag_run.run_id} and dag id {dag_run.dag_id}"
     )
+    synthetic_mats = []
+    # Peered dag-level materializations will always be emitted.
+    synthetic_mats.extend(synthetic_mats_for_peered_dag_asset_keys(dag_run, airflow_data))
+    # If there is a dagster run for this dag, we don't need to synthesize materializations for mapped dag assets.
```

Comment:
The double negative in the comment makes this a little hard to follow.

# No dagster runs means that the computation that materializes the asset was not proxied to Dagster. Therefore we are going to emit synthetic materializations based on the airflow executoion data.